### PR TITLE
chore(pml-analytics): Add analytics events to PML flow

### PR DESCRIPTION
**Pull request by jaredvu**
Add analytics for PML flow to study feature popularity/engagement.

---

## Views

- `<NewMarketForm>`
  - add `LaunchMarketFormStepChange` tracking as user progresses through flow

- `<NewMarketSelectionStep>`
  - add `LaunchMarketFormSelectedAsset` tracking when user selects an asset
 
- `<LaunchableMarket>`
  - add `LaunchMarketViewFromTradePage` tracking views

- `<LaunchableMarketChart>`
  - add `LaunchMarketPageChangePriceChartTimeframe` tracking timeframe changes

## Constants/Types

- `constants/analytics`
  - Add 5 new AnalyticsEvents around PML flow

## Hooks

- `hooks/usePotentialMarketsData`
  - update `hasPotentialMarketsData` to include testFlag for pml boolean

- `hooks/useSubaccount`
  - add  track for `LaunchMarketTransaction`
 
---

### CODE REVIEW

File: src/hooks/useSubaccount.tsx [hunk 1]
```diff
@@ -746,7 +746,15 @@ const useSubaccountContext = ({ localDydxWallet }: { localDydxWallet?: LocalWall
         throw new Error('wallet not initialized');
       }
 
-      const response = await compositeClient.createMarketPermissionless(subaccountClient, ticker);
+      track(AnalyticsEvents.LaunchMarketTransaction({ marketId: ticker }));
```
Comment (tyleroooo) [inline src/hooks/useSubaccount.tsx:749]: will errors thrown by this function be logged to datadog or anything?

File: src/views/charts/LaunchableMarketChart.tsx [hunk 2]
```diff
@@ -143,6 +145,20 @@ export const LaunchableMarketChart = ({
     );
   };
 
+  const trackTimeframeChange = useCallback(
+    (_timeframe: string) => {
```
Comment (tyleroooo) [inline src/views/charts/LaunchableMarketChart.tsx:149]: why _ before the name?

**Review by tyleroooo — APPROVED** [review]

File: src/hooks/useSubaccount.tsx (hunk 1 above)
Comment (jaredvu) [reply src/hooks/useSubaccount.tsx:749]: Yeah, it will be logged to bugsnag

File: src/views/charts/LaunchableMarketChart.tsx (hunk 2 above)
Comment (jaredvu) [reply src/views/charts/LaunchableMarketChart.tsx:149]: Ah I'll update this to `selectedTimeframe`. Originally I just didn't want it to clash with `timeframe` from `useState` regardless of scope

File: src/hooks/useSubaccount.tsx (hunk 1 above)
Comment (jaredvu) [reply src/hooks/useSubaccount.tsx:749]: datadog is built into the `log` function so it'll be sent to both

Comment (jaredvu) [reply src/hooks/useSubaccount.tsx:749]: Screen Shot 2024-10-30 at 9 41 44 AM

The dispatch will trigger bugsnag and the dd.error will also be sent

**Pull request merged**